src/lib/core/media-trigger/media-trigger.ts

```diff
@@ -0,0 +1,183 @@
+  // Internal Methods
+  // ************************************************
+
+  /**
+   * Whenever window resizes, immediately auto-restore original
+   * activations (if we are simulating activations)
+   */
+  private prepareAutoRestore() {
+    if (isPlatformBrowser(this._platformId) && this._document) {
+      const resize$ = fromEvent(window, 'resize').pipe(take(1));
+      this.resizeSubscription = resize$.subscribe(this.restore.bind(this));
+    }
+  }
+
+  /**
+   * Notify all matchMedia subscribers of de-activations
+   *
+   * Note: we must force 'matches' updates for
+   *       future matchMedia::activation lookups
+   */
+  private deactivateAll() {
+    const list = this.currentActivations;
+
+    this.forceRegistryMatches(list, false);
+    this.simulateMediaChanges(list, false);
+  }
+
+  /**
+   * Cache current activations as sorted, prioritized list of MediaChanges
+   */
+  private saveActivations() {
+    if (!this.hasCachedOriginals) {
+      const toMediaChange = (query: string) => new MediaChange(true, query);
+      const mergeMQAlias = (change: MediaChange) => {
+        let bp: OptionalBreakPoint = this.breakpoints.findByQuery(change.mediaQuery);
+        return mergeAlias(change, bp);
+      };
+
+      this.originalRegistry = this.matchMedia['registry'] as Map<string, MediaQueryList>;
+      this.originalActivations = this.currentActivations
+          .map(toMediaChange)
+          .map(mergeMQAlias)
+          .sort(sortDescendingPriority);
+
+      this.hasCachedOriginals = true;
+    }
+  }
+
+  /**
+   * Force set manual activations for specified mediaQuery list
+   */
+  private setActivations(list: string[]) {
+    if (!!this.originalRegistry) {
+      this.forceRegistryMatches(list, true);
+    }
+    this.simulateMediaChanges(list);
+  }
+
+  /**
+   * For specified mediaQuery list manually simulate activations or deactivations
+   */
+  private simulateMediaChanges(queries: string[], matches = true) {
+    const toMediaQuery = (query: string) => {
+      const locator = this.breakpoints;
+      const bp = locator.findByAlias(query) || locator.findByQuery(query);
+      return bp ? bp.mediaQuery : query;
+    };
+    const emitChangeEvent = (query: string) => this.emitChangeEvent(matches, query);
+
+    queries.map(toMediaQuery).forEach(emitChangeEvent);
+  }
+
+  /**
+   * Replace current registry with simulated registry...
+   * Note: this is required since MediaQueryList::matches is 'readOnly'
+   */
+  private forceRegistryMatches(queries: string[], match: boolean) {
+    const registry = new Map<string, MediaQueryList>();
+    queries.forEach(query => {
+      registry.set(query, {matches: match} as MediaQueryList);
+    });
+
+    this.matchMedia.registry = registry;
+  }
+
+  /**
+   * Restore original, 'true' registry
+   */
+  private restoreRegistryMatches() {
+    this.matchMedia.registry = this.originalRegistry as Map<string, MediaQueryList>;
+    this.originalRegistry = null;
```

Comment:
Instead of making the `Map` nullable, can we just call `this.originalRegistry.clear` here? That way we don't need to coerce types elsewhere.

Reply:
Fixed.